synfig-studio/src/gui/render.cpp

```diff
@@ -339,39 +371,100 @@ RenderSettings::on_render_pressed()
 	{
 		canvas_interface_->get_ui_interface()->error(_("A filename is required for this target"));
-		return;
+		return false;
 	}
-
-	hide();
-
-	render_passes.clear();
-	if (toggle_extract_alpha.get_active())
+	
+	String extension(filename_extension(filename));
+	bool ext_multi_file = false; //output target is an image sequence
+	int n_frames_overwrite = 0;
+	
+	//Retrieve current render settings
+	RendDesc rend_desc(widget_rend_desc.get_rend_desc());
+	
+	if(!toggle_single_frame.get_active() && (calculated_target_name != "png-spritesheet")
+			&& (rend_desc.get_frame_end() - rend_desc.get_frame_start()) > 0)
 	{
-		String filename_alpha(filename_sans_extension(filename)+"-alpha"+filename_extension(filename));
-
-		render_passes.push_back(make_pair(TARGET_ALPHA_MODE_EXTRACT, filename_alpha));
-		render_passes.push_back(make_pair(TARGET_ALPHA_MODE_REDUCE, filename));
+		//Check format which could have an image sequence as output
+		//If format is selected in comboboxtext_target
+		std::map<std::string,std::string> ext_multi = {{"bmp",".bmp"},{"cairo_png",".png"},
+					{"imagemagick",".png"}, {"jpeg",".jpg"},{"mng",".mng"},
+					{"openexr",".exr"},{"png",".png"},{"ppm",".ppm"}};
+	
+		std::map<std::string,std::string>::iterator ext_multi_it;
+		ext_multi_it = ext_multi.find(calculated_target_name);
+	
+		//calculated_target_name is a candidate with known output target (not Auto)
+		if(ext_multi_it != ext_multi.end())
+		{
+			extension = ext_multi_it->second;
+			ext_multi_file = true;
+		}
+		//otherwise Auto is selected
+		else
+		{
+			std::list<std::string> ext_multi_auto = {{".bmp"}, {".png"},
+					{".jpg"},{".exr"},{".ppm"}};
+	
+			ext_multi_file = (find(ext_multi_auto.begin(), ext_multi_auto.end(),
+					filename_extension(filename)) != ext_multi_auto.end());
+		}
 
-	} else {
-		render_passes.push_back(make_pair(TARGET_ALPHA_MODE_KEEP, filename));
+		//Image sequence: filename + sequence_separator + time
+		if(ext_multi_file)
+			for(int n_frame = rend_desc.get_frame_start();
+					n_frame <= rend_desc.get_frame_end();
+					n_frame++)
+			{
+				if(Glib::file_test(filename_sans_extension(filename) +
+					tparam.sequence_separator + 
+					etl::strprintf("%04d", n_frame) +
+					extension, Glib::FILE_TEST_EXISTS))
+					n_frames_overwrite++;
+			}
 	}
+
+	String message;
+	String details;
 	
-	App::dock_info_->set_n_passes_requested(render_passes.size());
-	App::dock_info_->set_n_passes_pending(render_passes.size());
-	App::dock_info_->set_render_progress(0.0);
-	App::dock_manager->find_dockable("info").present(); //Bring Dock_Info to front
+	if(n_frames_overwrite == 0)
+	{
+		message = strprintf(_("A file named \"%s\" already exists. "
+							"Do you want to replace it?"),
+							basename(filename).c_str());
 	
-	progress_logger->clear();
-	submit_next_render_pass();
+		details = strprintf(_("The file already exists in \"%s\". "
+							"Replacing it will overwrite its contents."),
+							basename(dirname(filename)).c_str());
```

Comment:
Used that because it's the same thing used for `.sif` files:
https://github.com/synfig/synfig/blob/b6995dc5864f43239b4e5ac9b119222328af6af1/synfig-studio/src/gui/instance.cpp#L545-L552

Reply:
Ouch.

Reply:
Maybe is better to use absolute path?

Reply:
If so, as it can be long, it would be better to place in a separated line.

Reply:
I fixed a couple of things but still there's this minor issue (IMHO that's why it's all on a single line):
![new_line_path](https://user-images.githubusercontent.com/6705690/83966267-72ffbc00-a8b9-11ea-84df-8518fe6b877b.png)
The text is centered. We could place it to the left somehow, but still in case of an RTL language it wouldn't be good looking either...

Reply:
btw if it's very long, it breaks the line on the slash without increasing the size of the dialog.